docs/xks/developer-guide/scheduling-scaling.md

```diff
@@ -0,0 +1,254 @@
+
+Most Kubernetes users encounter the scheduler for the first time when creating either a Deployment or Pod. The scheduler has to decide which Node to assign the Pod to. In the case of a Deployment the replica count can be increased and decreased at a moments notice by the user. When replica is changed for example from three to twenty the scheduler will have a lot of Pods all of a sudden to assign to Nodes. Having a deeper understanding of the Kubernetes scheduler will allow for a more efficient usage of resource consumption and debugging of any scheduler issues. A static replica count is the simplest method to work with Pod scaling and scheduling, this section will cover a lot of other options that exist in Kubernetes related to scaling and scheduling.
+
+## Pod Resources
+
+Setting the correct [resources for a Pod](https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/) is an important component in helping the scheduler select the correct Node to schedule a Pod to. There are two types of resource configuration fields, and these are configured for each container in a Pod.
+The resource request tells the scheduler how much of each resource each container in a Pod is expected to consume, it is however allowed of a Pod to consume more. The limit sets the maximum amount of resources each container in a Pod can consume. There are two main resource types that one should be aware of, CPU and memory.
+The CPU resource is defined in CPU units where one CPU unit is the equivalent of one CPU core. Fractional units can also be requested, in either decimals like `0.1` but also in terms of millicores where `100m` would be the same amount. The memory resource is defined in whole integers with the quantity suffixes
+`Ei, Pi, Ti, Gi, Mi, Ki`.
+
+If you do not specify any resources for a container the default resource request and limit will be applied as shown below. These resources are low on purpose, both to minimize the effects of over provisioning but also to make it obvious for end users that resources have not been specified. The keen-eyed may notice that
+a CPU limit is not set by default, this is on purpose and the reasons for it will be discussed later.
+
+```yaml
+resources:
+  request:
+    cpu: 50m
+    memory: 32Mi
+  limits:
+    memory: 256Mi
+```
+
+The scheduler will look at the cumulative resource requests across all containers in a Pod during the scheduling phase. The scheduler will exclude any Node which does not have capacity for the Pods resource request. Capacity is determined based on the total resources available in a Node minus the sum of all the
+resource requests of all Pods currently scheduled to the Node. This makes it possible to over provision as the resource request can be much larger than the actual consumption of the Pod. Efficient resource allocation is a constant battle between requesting enough resources to avoid under allocation while not

```

Comment:
I'm a bit confused about "over provision as the resource request can be much larger than the actual consumption of the Pod". Isn't over-provisioning the difference between request and limit?

Reply:
I misunderstood the middle sentence. On second reading, I think this is fine.